cmd/snap/cmd_sign.go

```diff
@@ -100,9 +105,54 @@ func (x *cmdSign) Execute(args []string) error {
 		return err
 	}
 
-	_, err = Stdout.Write(encodedAssert)
+	outBuf := bytes.NewBuffer(nil)
+	enc := asserts.NewEncoder(outBuf)
+
+	err = enc.WriteEncoded(encodedAssert)
+	if err != nil {
+		return err
+	}
+
+	if x.Chain {
+		accountKey, err := mustGetOneAssert("account-key", map[string]string{"public-key-sha3-384": privKey.PublicKey().ID()})
+		if err != nil {
+			return err
+		}
+
+		err = enc.Encode(accountKey)
+		if err != nil {
+			return err
+		}
+
+		account, err := mustGetOneAssert("account", map[string]string{"account-id": accountKey.(*asserts.AccountKey).AccountID()})
+		if err != nil {
+			return err
+		}
+
+		err = enc.Encode(account)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = Stdout.Write(outBuf.Bytes())
 	if err != nil {
 		return err
 	}
 	return nil
 }
+
+// call this function in a way that is guaranteed to specify a unique assertion
+// (i.e. with a header specifying a value for the assertion's primary key)
+func mustGetOneAssert(assertType string, headers map[string]string) (asserts.Assertion, error) {
+	asserts, err := downloadAssertion(assertType, headers)
+	if err != nil {
+		return nil, fmt.Errorf(i18n.G("cannot create assertion chain: %w"), err)
+	}
+
+	if len(asserts) == 1 {
```

Comment:
(super nitpick) I would invert the logic here, check for error here, then return the valid condition at the end. It seems to be the more common pattern in go, first deal with the errors, then with the "valid" data. I.e.:
```go
func mustGetOneAssert(assertType string, headers map[string]string) (asserts.Assertion, error) {
	asserts, err := downloadAssertion(assertType, headers)
	if err != nil {
		return nil, fmt.Errorf(i18n.G("cannot create assertion chain: %w"), err)
	}
	if len(asserts) != 1 {
		return nil, fmt.Errorf(i18n.G("internal error: cannot identify unique %s assertion for specified headers"), assertType)
	}

	return asserts[0], nil
}
```

Reply:
No argument, makes sense! Thanks